include/openPMD/backend/BaseRecord.hpp

```diff
@@ -181,6 +184,36 @@ BaseRecord< T_elem >::erase(key_type const& key)
     return res;
 }
 
+template< typename T_elem >
+inline typename BaseRecord< T_elem >::iterator
+BaseRecord< T_elem >::erase(iterator res)
+{
+    bool scalar = (res->first == RecordComponent::SCALAR);
```

Comment:
maybe just calling `erase(res->first)` (after checking `res` is not `end()`) is easier and avoids code duplication.